### DISCUSSION TITLE
mappable "push to mute" button for new video calls

### DISCUSSION THREAD

**Discussion by dull-kiwi** [element-hq/element-meta#2367] [Ideas]
Add a mappable 'push-to-mute' button for the new video rooms. This would mute the user in the call for as long as they held the button. Once the button is released they would become unmuted again.

This may seem niche but it's a great quality of life feature missing from element (and other matrix clients) preventing my friends from switching over.

The main use case is for playing games while on a matrix call, you may want to occasionally **_talk 'in-game' using a 'push-to-talk'_** button in-game. Mapping a _**'push-to-mute' function to the SAME button in element**_ avoids the user being heard by other call participants _twice_ - once in the call, and once in game (which never really sync up making it difficult to understand them and sometimes quite irritating to listen to).

**Comment by olfMombach** [comment]
From what I've read in [here](https://github.com/element-hq/element-web/issues/5993), for whatever reason the element team decided to completely disregard the functionality of push to talk or similar (like your proposal) **_since 2018_**, so I am not sure how realistic the implementation of this is.

It appears as though the main issue is a global keyboard shortcut provider library (or lack thereof), which the element team cannot devote the resources to developing (which is understandable). However from what official members have commented in the thread, there also seems to be the general perception of the team that none of this functionality is needed or in their core focus:
> Our efforts are prioritised on our target markets which currently does not include being a Discord replacement, doesn't mean it can't be one, just unlikely to see it on any roadmaps soon given focus for this calendar year is elsewhere

...which was from 3 weeks ago. Bummer
